### PR TITLE
Advanced Scroll Animations

### PR DESCRIPTION
**Pull request by guysopher**

### CODE REVIEW

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 1]
```diff
@@ -831,6 +818,7 @@ export class GalleryContainer extends React.Component {
             this.props.options[optionsMap.behaviourParams.gallery.layoutDirection] ===
             GALLERY_CONSTS[optionsMap.behaviourParams.gallery.layoutDirection].RIGHT_TO_LEFT
           }
+          infiniteScrollAnimation={this.props.options[optionsMap.behaviourParams.gallery.infiniteScrollAnimation]}
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js:821]: should it pass as props?

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 2]
```diff
@@ -887,9 +876,9 @@ export class GalleryContainer extends React.Component {
           {(this.layoutCss || []).filter(Boolean).map((css, idx) => (
             <style id={`layoutCss-${idx}`} key={`layoutCss-${idx}`} dangerouslySetInnerHTML={{ __html: css }} />
           ))}
-          {(this.scrollCss || []).filter(Boolean).map((css, idx) => (
+          {/* {(this.scrollCss || []).filter(Boolean).map((css, idx) => (
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js:879]: remove comment

File: packages/gallery/src/components/gallery/proGallery/galleryScrollIndicator.js [hunk 3]
```diff
@@ -50,34 +68,43 @@ export default class ScrollIndicator extends React.Component {
...
         ) {
           this.setState({
-            scrollTop: left,
-            scrollLeft: left,
+            scrollingHorizontally: true,
+            scrollLeft: Math.round(left),
           });
+          this.pauseHorizontalScrolling();
           this.props.getMoreItemsIfNeeded(left);
           this.debouncedOnScroll({ left });
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/galleryScrollIndicator.js]: extract out of the minScrollPosition

File: packages/gallery/src/components/gallery/proGallery/galleryScrollIndicator.js [hunk 4]
```diff
@@ -105,6 +137,7 @@ export default class ScrollIndicator extends React.Component {
           });
           this.props.getMoreItemsIfNeeded(top);
         }
+        this.pauseVerticalScrolling();
         this.debouncedOnScroll({ top });
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/galleryScrollIndicator.js]: extract out of minScrollPosition

File: packages/gallery/src/components/gallery/proGallery/galleryView.js [hunk 5]
```diff
@@ -212,6 +212,8 @@ class GalleryView extends React.Component {
       container: this.props.container,
       options: this.props.options,
       settings: this.props.settings,
+      scrollAnimationCss: this.props.scrollAnimationCss,
+      domId: this.props.domId,
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/galleryView.js]: Why is needed

Comment (guysopher) [reply packages/gallery/src/components/gallery/proGallery/galleryView.js]: use galleryId

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 6]
```diff
@@ -586,13 +586,15 @@ class SlideshowView extends React.Component {
       container: this.props.container,
       options: this.props.options,
       settings: this.props.settings,
+      scrollAnimationCss: this.props.scrollAnimationCss,
+      noFollowForSEO: this.props.noFollowForSEO,
+      domId: this.props.domId,
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: use galleryId

File: packages/gallery/src/components/helpers/scrollHelper.js [hunk 7]
```diff
@@ -235,19 +238,34 @@ function horizontalCssScrollTo({
...
-    from,
-    to,
-    duration,
-    isRTL,
-    slideTransition,
-    isContinuousScrolling,
-    autoSlideshowContinuousSpeed,
-    currentScrollEndTimeout,
-  };
+  return new Promise((resolve) => {
```
Comment (guysopher) [inline packages/gallery/src/components/helpers/scrollHelper.js]: check this part again - if the change is not necessary remove it

File: packages/gallery/src/components/item/itemView.js [hunk 8]
```diff
@@ -389,7 +390,7 @@ class ItemView extends React.Component {
           handleItemMouseUp: this.handleItemMouseUp,
         }}
         hasLink={this.itemHasLink()}
-        isCurrentHover={this.simulateHover()}
+        // isCurrentHover={this.simulateHover()}
```
Comment (guysopher) [inline packages/gallery/src/components/item/itemView.js]: do not change this

File: packages/gallery/src/components/item/itemView.js [hunk 9]
```diff
@@ -1019,19 +1020,21 @@ class ItemView extends React.Component {
...
-        <div key={'item-container-div-' + id} {...elementProps}>
-          {innerDiv}
-        </div>
-      );
-    }
+
+    const hasLink = linkParams?.href?.length > 0;
+
+    const itemProps = {
+      key: 'item-container-link-' + id,
```
Comment (guysopher) [inline packages/gallery/src/components/item/itemView.js]: link / div

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js (hunk 1 above)
Comment (guysopher) [reply packages/gallery/src/components/gallery/proGallery/galleryContainer.js:821]: yes, this element doesn't get the options

**Comment by denzelford** [comment]
Advanced scroll animations add dynamic[,](https://topfollowapko.net/top-follow-apk-old-version/) visually appealing effects to web pages, enhancing user engagement and creating a more interactive browsing experience.

**Pull request closed without merging**